java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -451,21 +440,6 @@ class ReferenceModeStore private constructor(
         return@fn true
     }
 
-    @FlowPreview
-    private val clearStoreCachesFlow = combine(
```

Comment:
I think this logic is needed, we want to keep it (it's cleaning up backing store when  receiveQueue and callbacks are empty). I don't know if it needs that context or it can run in same context as the store.

Reply:
This method was marked as unused, and removing it didn't cause any build failures. Is something missing? Am I missing something? How was it being invoked?

Reply:
the val is not used, but the whole block of code is launching the consuming of the two flows right? maybe it can be done in an init block instead than a val, not sure, but we still want to launch the combine. Check with Ian as he wrote this code.

Reply:
if you remove it, hopefully some test would fail?

Reply:
All of the tests are passing :(

Reply:
Hmmm.... Is it exercised by a test that's currently ignored?

Reply:
I'll work on adding this back, and see if I can improve the testing situation.

Reply:
I had the same first thought cleaning up the coroutines for last launch. So I think we should make this part more explicit, i.e. this part is needed for cleaning up.

Also, what happens if the clearCoroutineContext is shutdown before this part is done? Is there any harmful side effect?

Reply:
Does it make sense to have a global coroutine context to run all this clean ups?

Reply:
Is there any reason this can't be run in the existing coroutine context that's already driving the message queue?

Instead of using flows, why not just add an `onQueueEmpty` callback to `MessageQueue` that accepts a `suspend` function. Then we can do something like:

receiveQueue.onEmpty {
  if (callbacks.isEmpty) {
    backingStore.clearStoresCache()
  }
}
```

The intent of this feels much clearer to me...

Reply:
Nevermind, this misses when `off` is called and the queue is already empty.

Reply:
Having to provide a global coroutineContext (or any context at all) to enable the cleanup feels like a warning flag to me; I want to think through this a bit more.

Reply:
I think some @jasonwyatt expertise would help me out here :-)

Reply:
@ianull

Reply:
This *cannot* be removed for memory footprint sake.
The `.launchIn(CoroutineScope(clearCoroutineContext + Job()))` would start flow collection in the given context when initializing the `val clearStoreCachesFlow`

Reply:
I think it makes a lot of sense to move this into a method (instead of being a val, and initiating the flow's collection implicitly in the constructor)

That said, I think if it's going to be a suspend function, rather than `.onEach {}.launchIn(..)` as you've got it in the latest version, it would make the most sense to just collect the flow (which will suspend the coroutine).  The main difference there, however, would be that the `create` method wouldn't return until this method finishes.  Also, I'm somewhat unclear as to why this method is called during `create`.

```kotlin
combine(...)
    .filter { ... }
    .collect {
        // ... the logic checking the receive queue's size and closing things 
    }
```

Reply:
I moved to calling the method during create, because `create` is already in a coroutine, so there's an existing coroutine context that can be used to launch the flow. You're right, it makes sense to just switch to using collect, in this case.

Is there any issue with the store clean up occurring on the same coroutine that store creation occurred on to me?

Reply:
I'm putting this PR on hold, and opening another one with tests for the cleanup flow.

Reply:
Added a test for this behavior to this PR.